tcblib/slack_functions.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+import hmac
+import slack
+
+def validate_request(req):
+    """Validate slack request to prevent the hax."""
+    slack_signing_secret = boto3.client('ssm').get_parameter(
+        Name='/slack/tcb/signing_secret',
+        WithDecryption=True
+    )['Parameter']['Value']
+    signing_secret = str.encode(slack_signing_secret)
+
+    slack_request_timestamp = req['headers']['X-Slack-Request-Timestamp']
+    request_body = req['body']
+
+    token = 'v0:{0}:{1}'.format(
+        slack_request_timestamp,
+        request_body
+    )
+    encoded_token = str.encode(token)
+
+    slack_signature = req['headers']['X-Slack-Signature']
+
+    encrypted_secret = hmac.new(
+        key=signing_secret,
+        msg=encoded_token,
+        digestmod='sha256'
+    ).hexdigest()
+
+    expected_signature = 'v0={0}'.format(encrypted_secret)
+
+    if not hmac.compare_digest(expected_signature, slack_signature):
+        return False
+
+    return True
+
+def send_message(user_id, msg):
+    """DM the user because Slack changed bot permissions and it's the easiest way to go about it."""
```

Comment:
Yeah okay so originally I wanted to be cool and have the ephemeral message come back to you when you do stuff, like wherever you are, but new Slack bot permissions don't let you do that. This bot cannot post in any channel it is not explicitly in, and that includes ephemeral messages, so while having the bot DM you to confirm your star was added or to give you the host info you asked for kinda sucks, it's the only reasonable way to handle being able to support the slash commands in any of our like, 600 channels.